mtags-shared/src/main/scala/scala/meta/internal/pc/ConvertToNamedArgumentsUtils.scala

```diff
@@ -0,0 +1,26 @@
+package scala.meta.internal.pc
+
+import scala.annotation.tailrec
+
+class ConvertToNamedArgumentsUtils(text: String) {
+  @tailrec
+  final def findActualArgEnd(argEnd: Int, argsEnd: Int): Int =
+    if (argEnd == argsEnd || text.charAt(argEnd) == ',') argEnd
+    else findActualArgEnd(argEnd + 1, argsEnd)
+
+  /**
+   * Accounts for additional parenthesis around arguments:
+   *  e.g. f((a -> b))
+   */
+  def findActualArgBeginning(prevArgEnd: Int, argBeg: Int): Int = {
+    val inferredArgBeg = prevArgEnd + 1
+    if (inferredArgBeg == argBeg) argBeg
+    else {
+      val slice =
+        text.drop(inferredArgBeg).take(argBeg - inferredArgBeg)
+      val parenIndex = slice.indexOf('(')
```

Comment:
Could we also have `{` ?

Reply:
Actually, I just checked and If you do:
```
class C {
  def f(o: Int) = 1
  val g = f({2})
}
```
Converting to named args won't work for this argument at all. Seems like a whole separate issue.

Reply:
We can do a follow up for sure.